File: main/java/org/jskat/gui/table/OpponentPanel.java
package org.jskat.gui.table;

import javax.swing.ActionMap;

/**
 * Panel for showing informations about opponents
 */
public class OpponentPanel extends AbstractHandPanel {

	private static final long serialVersionUID = 1L;

	/**
	 * @see AbstractHandPanel#AbstractHandPanel(ActionMap, int, boolean)
	 */
	public OpponentPanel(ActionMap actions, int maxCards, boolean showIssWidgets) {

		super(actions, maxCards, showIssWidgets);
	}
}

File: jskat-gui/src/main/kotlin/org/jskat/gui/javafx/table/OpponentPanel.kt
package org.jskat.gui.javafx.table

class OpponentPanel(
    maxCards: Int,
    showIssWidgets: Boolean
) : AbstractHandPanel(maxCards, showIssWidgets) {

    init {
        // Allow the OpponentPanel itself to shrink below its content's minimum size.
        minHeight = 0.0
    }

    override fun createCardPanel(): CardPanel {
        val panel = super.createCardPanel()
        // CardPanel now handles its own clipping and minHeight in its init.
        return panel
    }
}
